File: rust/src/mark.rs
use automerge as am;

use crate::{am_value, bindings::Mark};

pub(crate) fn mark_to_java<'local>(
    env: &mut jni::Env<'local>,
    mark: &am::marks::Mark,
) -> jni::errors::Result<Mark<'local>> {
    let value = am_value::scalar_to_amvalue(env, mark.value())?;
    let name = env.new_string(mark.name())?;
    Mark::new(env, mark.start as i64, mark.end as i64, &name, &value)
}

File: lib/src/main/java/org/automerge/Mark.java
package org.automerge;

public class Mark {
	private final long start;
	private final long end;
	private final String name;
	private final AmValue value;

	protected Mark(long start, long end, String name, AmValue value) {
		this.start = start;
		this.end = end;
		this.name = name;
		this.value = value;
	}

	public long getStart() {
		return start;
	}

	public long getEnd() {
		return end;
	}

	public String getName() {
		return name;
	}

	public AmValue getValue() {
		return value;
	}

	@Override
	public String toString() {
		return "Mark [start=" + start + ", end=" + end + ", name=" + name + ", value=" + value + "]";
	}
}
